das/daser_test.go

```diff
@@ -274,20 +274,23 @@ func TestDASer_stopsAfter_BEFP(t *testing.T) {
 	// create fraud service and break one header
 	f := fraud.NewService(ps, mockGet.GetByHeight, ds)
 	mockGet.headers[1] = header.CreateFraudExtHeader(t, mockGet.headers[1], bServ)
-
+	newCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
 	// create and start DASer
 	daser := NewDASer(shareServ, sub, mockGet, ds, f)
-
-	require.NoError(t, daser.Start(ctx))
-	fraud.OnProof(ctx, f, fraud.BadEncoding,
+	ch := make(chan struct{})
+	go fraud.OnProof(newCtx, f, fraud.BadEncoding,
 		func(proof fraud.Proof) {
-			require.NoError(t, daser.Stop(ctx))
+			ch <- struct{}{}
 		})
-	// wait for dasing catch-up routine fails
+	time.Sleep(time.Millisecond * 100)
+	require.NoError(t, daser.Start(newCtx))
+	// wait for fraud proof will be handled
 	select {
 	case <-ctx.Done():
 		t.Fatal(ctx.Err())
-	case <-daser.ctx.Done():
+	case <-ch:
+		require.NoError(t, daser.Stop(ctx))
```

Comment:
Doesn't this not actually test that the daser is stopped by the BEFP handling?

Reply:
reworked